### ISSUE TITLE
Feature: Use icons from manifest.json when WebApps provide one.

### ISSUE THREAD

**Issue by panaaj** [SignalK/signalk-server#629]
`manifest.json` is used by modern web apps (including PWAs) to describe how the application should be presented on a device.
Some of the information contained in `manifest.json` does overlap with `package.json` but it does provide additional information that could enhance the presentation of the `WebApps` list in the server admin-ui.
i.e. displaying the app icon instead of a generic icon.

Example `manifest.json` content for Freeboard follows:
```{
  "name": "Freeboard",
  "short_name": "Freeboard",
  "theme_color": "#1976d2",
  "background_color": "#fafafa",
  "display": "fullscreen",
  "orientation": "portrait",
  "scope": "/@signalk/freeboard-sk/",
  "start_url": "index.html",
  "icons": [
    {
      "src": "assets/icons/icon-72x72.png",
      "sizes": "72x72",
      "type": "image/png"
    },
    {
      "src": "assets/icons/icon-96x96.png",
      "sizes": "96x96",
      "type": "image/png"
    },
    {
      "src": "assets/icons/icon-128x128.png",
      "sizes": "128x128",
      "type": "image/png"
    },
    {
      "src": "assets/icons/icon-144x144.png",
      "sizes": "144x144",
      "type": "image/png"
    },
    {
      "src": "assets/icons/icon-152x152.png",
      "sizes": "152x152",
      "type": "image/png"
    },
    {
      "src": "assets/icons/icon-192x192.png",
      "sizes": "192x192",
      "type": "image/png"
    },
    {
      "src": "assets/icons/icon-384x384.png",
      "sizes": "384x384",
      "type": "image/png"
    },
    {
      "src": "assets/icons/icon-512x512.png",
      "sizes": "512x512",
      "type": "image/png"
    }
  ]
}```

**Comment by tkurki** [comment]
This works only for installed webapps - which is fine and implementable, but having per webapp icons also in the app store would be even cooler. 

That information would have to go into package.json though and the image must be hosted on the net (which by definition is available if you are dealing with app store).

**Comment by panaaj** [comment]
So would this mean the addition of a “signalk-appstore” or “signalk” section to the package.json for this and other possible uses.

**Comment by tkurki** [comment]
Something like that, yes.

**Comment by panaaj** [comment]
Reviving this issue in light of v2 work... there is an opportunity define a `signalk` section of the `package.json` file to hold attributes about the plugin / webapp to both enhance the presentation of installed applications but  to also filter the list of available apps to those that support the version of server being used.

For example:
```JSON
"signalk": {
   "displayName":  "Freeboard-SK", // Screen friendly name to display rather than npm package name i.e. @signalk/freeboard-sk)
   "appIcon": "./public/assets/icons/icon.png", // path to icon file when installed
   "minServerVersion": "2.0.0",  // minimum supported server version. Present only plugin / apps supporting the current server version.
   "appStoreIcon": "https://github.com/SignalK/freeboard-sk/tree/master/src/assets/icons/icon.png" // path to icon to display in app store
}
```
This is something that coulkd be implemented now with little implact but provide a smoother path to a v2 release.

**Comment by panaaj** [comment]
Addressed in commit [f716ac2](https://github.com/SignalK/signalk-server/commit/f716ac274c31646da8e1cdcb85b288ffc1b09aa1)
